core/modules/editor/factory.js

```diff
@@ -285,17 +285,19 @@ function editTextWidgetFactory(toolbarEngine,nonToolbarEngine) {
 	Propogate keydown events to our container for the keyboard widgets benefit
 	*/
 	EditTextWidget.prototype.propogateKeydownEvent = function(event) {
-		var newEvent = this.document.createEventObject ? this.document.createEventObject() : this.document.createEvent("Events");
-		if(newEvent.initEvent) {
-			newEvent.initEvent("keydown", true, true);
+		if(this.engine.constructor.name === "FramedEngine") {
```

Comment:
Apologies if I've missed something in the discussion, but why do we need to check explicitly for the framed engine?

Reply:
Hi @Jermolene - the propogateKeydownEvent method creates a new Event also in a CodeMirror instance and that new Event is because of the iframe of the framed engine not propagating the event up to the container by default. In the CodeMirror text editor a global keyboard shortcut would get dispatched twice, that's why I'm testing if we're within a FramedEngine here

Reply:
There are probably better fixes but this is the best I've come up with

Reply:
@Jermolene do you know a better fix?